torch/_dynamo/device_interface.py

```diff
@@ -68,23 +68,23 @@ class Worker:
         """
 
         @staticmethod
-        def set_device(device: int):
+        def set_device(device: int) -> NoReturn:
             raise NotImplementedError
```

Comment:
Are you sure this annotation is appropriate? I don't remember what NoReturn's subtyping relation is but it seems like it would be more useful for these to be annotated with the type that subclasses are expected to implement.